assets/styl/_modules/_form.styl

```diff
@@ -0,0 +1,60 @@
+// Forms widgets
+//
+// Colette provides styles for:
+// - Fields descriptions (with valid / invalid color indication)
+// - Valid / Invalid fields
+// - Also a special style for a flex parent with 3 children (label, infolink, input 100%)
+//
+// Note: you can also use Alerts components to handle success or errors messages.
+//
+// Weight: 2
+//
+// Markup: form.twig
+//
+// Styleguide: Components.Forms.Form
+.form
+    &-field-widget
+        // display a widget for form element that must be filled with text content
+        // e.g. input[text|password|email|url]
+
+        label
+            display block
+
+        ul
+            margin 0
+
+        //special style for a flex parent
+        &-flex
+            display flex
+            flex-wrap wrap
+
+            label
+                flex-grow 1
+                flex-basis auto
+
+            .form-infolink
```

Comment:
I think `form-infolink` can be a simple selector, and that this declaration can be at the same level as `&-field-widget`, `&-field-error`, etc (though maybe the padding-top could depend on the `&-flex` context).